core/dictgen/CMakeLists.txt

```diff
@@ -13,9 +13,13 @@ ROOT_GLOB_SOURCES(sources ${CMAKE_CURRENT_SOURCE_DIR}/src/*.cxx)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src
         ${CMAKE_CURRENT_SOURCE_DIR}/res
         ${CMAKE_CURRENT_SOURCE_DIR}/../foundation/res
-        ${CMAKE_CURRENT_SOURCE_DIR}/../clingutils/res
-        ${LLVM_INCLUDE_DIRS}
-        ${CLING_INCLUDE_DIRS})
+        ${CMAKE_CURRENT_SOURCE_DIR}/../clingutils/res)
+include_directories(${CLING_INCLUDE_DIRS})
+# Register the llvm include directories after clangs. This instructs the compiler to resolve
+# headers from our builtin clang. That's an issue when we are building with bultin_llvm=Off
+# and we have installed clang headers, too.
+include_directories(SYSTEM ${CLANG_INCLUDE_DIRS} ${LLVM_INCLUDE_DIRS})
```

Comment:
You may want to use `BEFORE` keyword when adding `${CLANG_INCLUDE_DIRS}`.

Reply:
I am not sure I get your idea. I need `CLANG_INCLUDE_DIRS` to show up before `LLVM_INCLUDE_DIRS`. IIUC, if I want to use `BEFORE` I will have to have at least two lines. Can you paste a code snippet here and I'll be happy to use it.